$MirOS$

commit a7e79f28689c574e0bbef17f4cb3da00249181ff
Author: Mark Adler <madler@alumni.caltech.edu>
Date:   Tue Jan 19 16:28:24 2010 +0100

    libxml violates the zlib interface and crashes
    
    * xmlIO.c: remove an abuse of zlib API and use a clean interface
      available in zlib >= 1.2.3

--- xmlIO.c.orig	Thu Feb 21 21:52:44 2008
+++ xmlIO.c	Wed Sep 11 16:43:47 2013
@@ -2405,6 +2405,9 @@ __xmlParserInputBufferCreateFilename(con
 #ifdef HAVE_ZLIB_H
 	if ((xmlInputCallbackTable[i].opencallback == xmlGzfileOpen) &&
 		(strcmp(URI, "-") != 0)) {
+#if defined(ZLIB_VERNUM) && ZLIB_VERNUM >= 0x1230
+	    ret->compressed = !gzdirect(context);
+#else
 	    if (((z_stream *)context)->avail_in > 4) {
 	        char *cptr, buff4[4];
 		cptr = (char *) ((z_stream *)context)->next_in;
@@ -2416,6 +2419,7 @@ __xmlParserInputBufferCreateFilename(con
 		    gzrewind(context);
 		}
 	    }
+#endif
 	}
 #endif
     }
